repos/spack_repo/builtin/packages/umpire/package.py

```diff
@@ -341,7 +349,7 @@ def initconfig_compiler_entries(self):
 
         option_prefix = "UMPIRE_" if spec.satisfies("@2022.03.0:") else ""
 
-        if spec.satisfies("+fortran") and compiler.fc is not None:
+        if spec.satisfies("+fortran") and "fortran" in spec:
```

Comment:
```suggestion
        if spec.satisfies("+fortran"):
```

When `+fortran` there will always be a fortran compiler so the second statement will always be a noop.